coverage/docs/design.md

```diff
@@ -0,0 +1,139 @@
+ recent successful post-submit build. Coverage file for post-submit commits were generated in 
+ post-submit workflow and stored in gcs bucket
+5. Use PR data from github, git-attributes, as well as coverage change data calculated above, to 
+produce a list of files that we care about in the line-by-line coverage report. produce line by 
+line coverage html and add link to covbot report. Note that covbot is the robot github account 
+used to report code coverage change results. See Covbot section for more details.
+6. Let covbot post presubmit coverage on github, under that conversation of the PR. 
+
+##Post-submit workflow
+Produces & stores coverage profile for later presubmit jobs to compare against; 
+Produces periodical coverage result as input for TestGrid. Testgrid can use the data produced here to display coverage trend in a tabular or graphical way. 
+1. A PR is merged
+2. Post-submit job started
+3. Generate coverage profile. Completion marker generated upon successful run. Both stored
+ in artifacts directory.
+    - Completion marker is used by later pre-submit job when searching for a healthy and complete 
+    code coverage profile in the post-submit jobs
+    - Successfully generated coverage profile may be used as the basis of comparison for coverage change, 
+    as mentioned in pre-submit workflow
+4. Generate / store per-file coverage data
+    - Stores in the XML format, that is used by TestGrid, and dump it in artifacts directory
+
+##Locally running presubmit and post-submit workflows
+Both workflows may be triggered locally in command line, as long as all the required flags are 
+supplied correctly. In addition, the following env var needs to be set:
+- JOB_TYPE (one of 'presubmit', 'postsubmit', 'periodic', 'local-presubmit')
+
+use 'local-presubmit' will run the presubmit workflow without posting result on github PR
+  
+##Covbot
+As mentioned in the presubmit workflow section, covbot is the short name for the robot github 
+account used to report code coverage change results. It can be created as a regular github 
+account - it does not need to be named covbot as that name is already taken on Github. It only need a 
+comment access to the repo it need to be run on. If the repo is private, it also need read access. 
+  
+After the robot account is created, download the github token and supply the path to the token 
+file to code coverage binary, as the value for parameter "github-token"
+
+#Usage with prow
+
+Prow can be used as the system to handle Github events mentioned in the two workflows. We can add a pre-submit prow job that is triggered by any new commit to a PR to run test coverage on the new build to compare it with the master branch and previous commit for pre-submit coverage. We can add a post-submit prow job that is triggered by merge events, to run test coverage when ever there is a merge on the target branch. 
+
+In addition, at the end of each workflow, prow copies the artifacts directory to gcs bucket
```

Comment:
If you are going to mention this you should also mention the other things Prow does for the job. Namely checking out the appropriate code, uploading logs, and reporting a pass or fail status context to GitHub.

This is also the section where you can mention that Tide can be configured to block merge if the job fails.

Reply:
Done.